tiledb/array.py

```diff
@@ -157,11 +159,40 @@ def index_domain_subarray(array, dom, idx: tuple):
         else:
             (dim_lb, dim_ub) = dim.domain
 
-        dim_slice = idx[r]
-        if not isinstance(dim_slice, slice):
-            raise IndexError("invalid index type: {!r}".format(type(dim_slice)))
+        dim_idx = idx[r]
+
+        if isinstance(dim_idx, np.ndarray) or isinstance(dim_idx, list):
+            if not np.issubdtype(dim_dtype, np.str_) and not np.issubdtype(
+                dim_dtype, np.bytes_
+            ):
+                # if this is a list, convert to numpy array
+                if isinstance(dim_idx, list):
+                    dim_idx = np.array(dim_idx)
+                subarray.append((dim_idx))
```

Comment:
Are you intending to create a tuple here?

```suggestion
                subarray.append((dim_idx,))
```